### ISSUE TITLE
** (UndefinedFunctionError) function Repo.transaction/1 is undefined (module Repo is not available)

### ISSUE THREAD

**Issue by seymores** [izelnakri/paper_trail#28]
I'm trying out the latest `0.7.6` and getting this error when trying to update a record.
The update is a simple ` PaperTrail.update(changeset, origin: "test") `

I did condigure the `config :paper_trail, repo: MyApp.Repo ` and seems to work when I call ` Papertrail.Version.count() `.

```
** (UndefinedFunctionError) function Repo.transaction/1 is undefined (module Repo is not available)
    Repo.transaction(%Ecto.Multi{names: #MapSet<[:model, :version]>, operations: [version: {:run, #Function<10.48676158/1 in PaperTrail.update/2>}, model: {:changeset, #Ecto.Changeset<action: :update, changes: %{name: "First Surname v3"}, errors: [], data: #App.Person<>, valid?: true>, []}]})
    (paper_trail) lib/paper_trail.ex:151: PaperTrail.update/2
```

From iex console, calling `PaperTrail.RepoClient.repo` returns the correct Repo.
So the configuration should not be the problem.

**Comment by izelnakri** [comment]
did you alias your MyApp.Repo in your mix application? Otherwise you should use `MyApp.Repo.update/2`

**Comment by izelnakri** [comment]
Which version of ecto and postgrex do you use?

**Comment by seymores** [comment]
More findings in iex console:

It looks like the `config :paper_trail, repo: TSC.Repo, item_type: Ecto.UUID, originator_type: Ecto.UUID` config is not loaded at runtime.
When I reload PaperTrail module using the `r PaperTrail`, my repo shows up correctly but then I encountered a different problem:

```
** (Ecto.ChangeError) value `"f75cff7e-4aeb-4fe8-a262-c9c4c00ea9b5"` for `PaperTrail.Version.item_id` in `insert` does not match type :integer
```

I am not sure why it is expecting `:integer` type for item_id even though my config specifies Ecto.UUID and the table generated is `:uuid`.

**Comment by seymores** [comment]
Never mind -- I have no fucking clue it's working perfectly fine after a mix clean.
Problem must be between the keyboard and chair. ;-)

**Comment by seymores** [comment]
Hi @izelnakri -- I have to reopen this ticket because the problem is back after a project clean and recompile.

I have tracked the problem to the use of module attribute '@' for referencing repo, item_type, and originator_type.

I am using Elixir 1.5 and not sure if the behavior of referencing using module attribute changed since 1.4, but I am getting consistent problems of the config not being picked up by PaperTrail now.

I forked a copy and the error is no more after replacing the use of @repo, @item_type, and @originator_type with direct function calls to obtain them.
These module attributes failed to get the correct values at runtime.

**Comment by backspace** [comment]
Hey, thanks for `PaperTrail`, @izelnakri, I’m glad I found it when I did because I was almost done a branch that incorporated an obscure fork of a seemingly-abandoned library!

And thanks for this thread, @seymores, because I was having this problem and #29 fixed it for me. I don’t know enough at this point to have any suggestions about whether it’s a proper fix, but it’s made `PaperTrail` look like it’ll work for me in a situation using `:binary_id` for primary keys.

I’m on `ecto` 2.2.1 and `postgrex` 0.13.3. The code is open source and linked above, but let me know if there’s anything else I can add that might help.
